### ISSUE TITLE
extra mutex in broadcast example

### ISSUE THREAD

**Issue by troppoli** [zaphoyd/websocketpp#617]
It seems to me that the m_connection_lock mutex in the broadcast_server isn't needed. Connections are only manipulated inside the process_messages loop serially. 

I think it's helpful to see what's minimally required for synchronization.

**Comment by zaphoyd** [comment]
Yes it looks like this is the case. Specifically if you only ever have one worker thread and only the worker thread ever updates the connection list then you don't need to lock the connection list itself.

That said, in most WebSocket++ programs that I've seen there are either multiple worker threads or both the main and worker threads inspect the connection list. I'll need to think a bit about exactly how that should be conveyed. At the very minimum I will add a comment explaining precisely when that particular mutex is required.

**Comment by troppoli** [comment]
Thanks. I spent time digging through the tests and samples when I was getting started with websocketpp... it's great to be able to say I understand every line in this sample when you start designing more elaborate uses. It's been a great library to work with.
